### PR TITLE
Implement print_table/2

### PR DESCRIPTION
**Pull request by noahshaw11**
## Overview

Implement `couch_debug:print_table/2` to facilitate memory leak detections.

## Checklist

- [x] Code is written and works correctly
- [ ] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/couch/src/couch_debug.erl [hunk 1]
```diff
@@ -273,6 +278,17 @@ couch_index_server[<0.288.0>]                |   478240   |         0         |
...
+        ---
+    ", []);
+help(print_table) ->
+    io:format("
+        print_table(Rows, TableSpec)
+        --------------------------------
+
+        Print table of specifications.
+          - Rows: List of {Id, Props} to be printed from the TableSpec
+          - TableSpec: List of either {Value} or {Width, Align, Value}.
```
Comment (iilyak) [inline src/couch/src/couch_debug.erl:290]: Where `Align` is ...

Comment (iilyak) [reply src/couch/src/couch_debug.erl:290]: left/center/right

File: src/couch/src/couch_debug.erl [hunk 2]
```diff
@@ -615,6 +632,16 @@ restart_busy(ProcessList, Threshold, DelayInMsec, Property) when
...
+    io:format("~s~n", [format(TableSpec)]),
+    lists:foreach(
+        fun({Id, Props}) ->
+            io:format("~s~n", [table_row(Id, 2, Props, TableSpec)])
+        end,
+        Rows
+    ),
+    ok.
+
 print_tree(Tree, TableSpec) ->
```
Comment (iilyak) [inline src/couch/src/couch_debug.erl:645]: both print_table and print_tree should have comments documenting their arguments.

Comment (noahshaw11) [reply src/couch/src/couch_debug.erl:657]: Added.

**Pull request merged**